File: perf_test/array/array.c
#include<cstdio>
#include<ctime>

#define ARR_LEN 1000000
#define READS 1000000

int x[ARR_LEN];
int reads[READS];

int main () {
  clock_t time = clock();
  x[0] = 19;
  for (int i = 1; i <= ARR_LEN; i++)
    x[i] = ((x[i - 1] * 23) + 53) % 22695477;


  reads[0] = 23;
  for (int i = 1; i < READS; i++)
    reads[i] = (((reads[i - 1] * 17) + 37) % 1664525) % 100000;

  for (int i = 0; i < READS; i++) {
    ++x[reads[i]];
  }
  printf("time: %ld %ld %lf\n", time, clock(), ((double)clock() - time)/CLOCKS_PER_SEC);
  // console.log(new Date() - time);
}

File: perf_test/array/array.js
var ARR_LEN = 1000000;
var READS = 1000000;

var time = new Date();
var x = [];
x[0] = 19;
for (var i = 1; i <= ARR_LEN; i++)
    x[i] = ((x[i - 1] * 23) + 53) % 22695477;

var time2 = new Date();

var reads = [];
reads[0] = 23;
for (i = 1; i < READS; i++)
    reads[i] = (((reads[i - 1] * 17) + 37) % 1664525) % 100000;

var time3 = new Date();

var a;
for (i = 0; i < READS; i++) {
    ++x[reads[i]];
}
console.log(time2 - time);
console.log(time3 - time2)
console.log(new Date() - time3);
